feat: :heavy_plus_sign: Adding export as image in profile page

dimassibassem opened the pull request:
Resolve #39 

This pull request introduces a feature to the /profile page – the addition of a sleek export button. This button empowers users to export an Image of the card of their profile.

![image](https://github.com/Balastrong/github-stats/assets/75867744/da1ff0f6-6191-4e15-b573-a2d9ba9fd39f)

Balastrong commented on src/utils/exportAsImage.ts:
```diff
@@ -22,7 +23,7 @@ export const exportStats = async (
           const a = document.createElement("a");
           a.href = url;
-          a.download = "stats.png";
+          a.download = `${filename}.png`;
```
`filename` is optional in the function here, we should either make sure that it mandatory when going with `option: "download"` or give it a default

Balastrong commented on src/pages/profile/index.tsx:
```diff
@@ -5,45 +5,89 @@ import {
+            <li>
+              <button
+                className="btn-ghost"
+                onClick={async () =>
+                  await exportAsImage(
```
I missed this on the other PR but I still think about it... do we really need to declare the handler `async` and then `await` when calling it?

I mean, we're not doing anything else after that, but I also get that it can be a good practice to keep it there.

I'm fine either way, just wanted to point it out to make sure we're doing it intentionally :)

Balastrong reviewed (changes requested):
Thank you for the PR! I added a couple tiny comments.

There's one extra thing though that we need to tackle, I see the download is much wider than it probably should

![image](https://github.com/Balastrong/github-stats/assets/7253929/9c12567f-564b-4e16-b245-ce67b7f11695)

Do you think we can find a way to actually export only the width of the card?

dimassibassem commented:
Hey @Balastrong, 

Thank you for your feedback on the pull request! I've reviewed your comments and incorporated your suggestions into the latest changes.

Regarding the width issue with the download, I've addressed this concern in the new commit. The download width should now be in line with the expected dimensions.
example: 
<img width='400px' src='https://github.com/Balastrong/github-stats/assets/75867744/883ac908-f58c-4968-95e7-e031ec5a4a1d' />

I've taken your suggestion into account about the filename in the function. In the new commit, it become this way
```javascript
a.download = filename ? `${filename}.png` : "image.png";
```
I've carefully considered your observation about the async/await handler. In response, I've removed the async/await structure for now. However, I want to clarify that the initial reason for its inclusion was to pave the way for potential notifications after the function's resolution in the future. Your attention to detail and thoughtful questions are greatly appreciated, as they prompted me to rethink the necessity in the present context. Please feel free to share any additional thoughts or suggestions you might have!

Balastrong commented:
Thank you for the changes, looks perfect now!

I also agree with the async/await explanation, it totally makes sense! Actually, we could already spawn an issue on the repo to implement our own toast system (to practice building one!) and when it's ready we can indeed put it after the download function inside the onClick :)

---

I see there's a little conflict on `src/pages/stats/[login].tsx` now, could you please fix it?

dimassibassem commented:
Hey @Balastrong, 
Should be resolved now

Balastrong reviewed (approved):
Thank you! 🚀
